### PR TITLE
Support putting attributes for dimensions & datatypes for ncvars

### PR DESCRIPTION
**Pull request by kouketsu**
"nccreate" didn't support to write attributes for dimensions & didn't support datatypes except "NC_DOUBLE". 
I try to fix those with this pull request.

### CODE REVIEW

File: src/NetCDF.jl [hunk 1]
```diff
@@ -249,7 +255,7 @@ for (t,ending,arname) in funext
     @eval nc_get_var1_x(ncid::Integer,varid::Integer,start::Vector{UInt},::Type{$t})=begin $fname1(ncid,varid,start,$(arsym)); $(arsym)[1] end
 end
 
-function nc_get_vara_x!{T<:Union{UTF8String,ASCIIString}}(ncid::Integer,varid::Integer,start::Vector{UInt},count::Vector{UInt},retvalsa::Array{T})
+function nc_get_vara_x!(ncid::Integer,varid::Integer,start::Vector{UInt},count::Vector{UInt},retvalsa::Array{AbstractString})
```
Comment (meggart) [inline src/NetCDF.jl]: Can you loosen this again by making the function parametric? This way an Array{UTF8String} would not be accepted by the signature, so we should use

    function nc_get_vara_x!{T<:AbstractString}(....retvalsa::Array{T})

File: src/NetCDF.jl [hunk 2]
```diff
@@ -259,7 +265,7 @@ function nc_get_vara_x!{T<:Union{UTF8String,ASCIIString}}(ncid::Integer,varid::I
   retvalsa
 end
 
-function nc_get_var1_x(ncid::Integer,varid::Integer,start::Vector{UInt},::Union{Type{ASCIIString},Type{UTF8String}})
+function nc_get_var1_x(ncid::Integer,varid::Integer,start::Vector{UInt},::Type{AbstractString})
```
Comment (meggart) [inline src/NetCDF.jl]: Same here, this would not work for String types since isa(Type{ASCIIString},Type{AbstractString}) returns false...

File: src/NetCDF.jl [hunk 3]
```diff
@@ -411,9 +419,9 @@ function create(name::AbstractString,varlist::Array{NcVar};gatts::Dict{Any,Any}=
 
   #Create the file
   id = nc_create(name,mode)
-
   # Collect Dimensions and set NetCDF ID
-  vars=Dict{ASCIIString,NcVar}();
+  # vars=Dict{ASCIIString,NcVar}();
+  vars = Dict{AbstractString,NcVar}();
```
Comment (meggart) [inline src/NetCDF.jl]: Why is this necessary? In general one should not use Abstract types in containers. Are there non-ASCII variable names? Would UTF8String be sufficient?

**Comment by meggart** [comment]
Thanks a lot for looking into this. Originally I did this because some NetCDF readers had problems interpreting non-DOUBLE dimension variables, so I set Float64 to default. However, it seems better to be not so strict and let the user decide...

**Comment by kouketsu** [comment]
Fix the changes pointed out by meggart. Sorry, as I leave changes not related with this PR aims, I might confuse.

**Comment by meggart** [comment]
Thanks for this I really like the new jl2nc function, makes things more generic.
